clang: error: no such file or directory: 'libclang_rt.fuzzer_no_main.a'

```
HEAD is now at aae4bbd cmake: introduce CMake option OSS_FUZZ
[48/66] cd /src/testdir/build/luzer/source && /usr/local/bin/cmake -Dcan_fetch=YES -P /src/testdir/build/luzer/tmp/bundled-luzer-gitupdate.cmake
[49/66] cd /src/testdir/build/tests/lapi && /usr/local/bin/cmake -E echo_append && /usr/local/bin/cmake -E touch /src/testdir/build/luzer/stamp/bundled-luzer-patch
[50/66] : && /usr/local/bin/clang++ -O1   -fno-omit-frame-pointer   -gline-tables-only   -Wno-error=incompatible-function-pointer-types   -Wno-error=int-conversion   -Wno-error=deprecated-declarations   -Wno-error=implicit-function-declaration   -Wno-error=implicit-int   -Wno-error=unknown-warning-option   -Wno-error=vla-cxx-extension   -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ -O1   -fno-omit-frame-pointer   -gline-tables-only   -Wno-error=incompatible-function-pointer-types   -Wno-error=int-conversion   -Wno-error=deprecated-declarations   -Wno-error=implicit-function-declaration   -Wno-error=implicit-int   -Wno-error=unknown-warning-option   -Wno-error=vla-cxx-extension   -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ tests/capi/CMakeFiles/lua_load_test.dir/lua_load_test.cc.o -o tests/capi/lua_load_test  lua-master/source/
[51/66] /usr/local/bin/clang++  -I/src/testdir/build/lua-master/source -I/src/testdir/build/tests/capi/external.protobuf_mutator/include -I/src/testdir/build/tests/capi/external.protobuf_mutator/include/libprotobuf-mutator -I/src/testdir/build/tests/capi -I/src/testdir/build/tests/capi/external.protobuf_mutator/src/external.protobuf_mutator-build/external.protobuf/include -O1   -fno-omit-frame-pointer   -gline-tables-only   -Wno-error=incompatible-function-pointer-types   -Wno-error=int-conversion   -Wno-error=deprecated-declarations   -Wno-error=implicit-function-declaration   -Wno-error=implicit-int   -Wno-error=unknown-warning-option   -Wno-error=vla-cxx-extension   -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ -Wall -Wextra -Wpedantic -Wno-unused-parameter -g -MD -MT tests/capi/CMakeFiles/torture_test.dir/torture_test.cc.o -MF tests/capi/CMakeFiles/torture_test.dir/torture_test.cc.o.d -o tests/capi/CMakeFiles/tortu
[52/66] cd /src/testdir/build/luzer/build && /usr/local/bin/cmake -B /src/testdir/build/luzer/build -S /src/testdir/build/luzer/source -G Ninja -DCMAKE_C_COMPILER=/usr/local/bin/clang -DCMAKE_CXX_COMPILER=/usr/local/bin/clang++ -DCMAKE_BUILD_TYPE= -DLUA_INCLUDE_DIR=/src/testdir/build/lua-master/source/ -DLUA_LIBRARIES=/src/testdir/build/lua-master/source/liblua.a -DOSS_FUZZ=ON && /usr/local/bin/cmake -E touch /src/testdir/build/luzer/stamp/bundled-luzer-configure
-- The C compiler identification is Clang 22.0.0
-- The CXX compiler identification is Clang 22.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- [SetClangRTLib] LIBCLANG_ASAN_LIB is /usr/local/lib/clang/22/lib/x86_64-unknown-linux-gnu/libclang_rt.asan.a
-- [SetClangRTLib] LIBCLANG_UBSAN_LIB is /usr/local/lib/clang/22/lib/x86_64-unknown-linux-gnu/libclang_rt.ubsan_standalone.a
-- [SetClangRTLib] FUZZER_NO_MAIN_LIBRARY is /usr/local/lib/clang/22/lib/x86_64-unknown-linux-gnu/libclang_rt.fuzzer_no_main.a
-- Configuring done (1.7s)
-- Generating done (0.0s)
-- Build files have been written to: /src/testdir/build/luzer/build
[53/66] : && /usr/local/bin/clang++ -O1   -fno-omit-frame-pointer   -gline-tables-only   -Wno-error=incompatible-function-pointer-types   -Wno-error=int-conversion   -Wno-error=deprecated-declarations   -Wno-error=implicit-function-declaration   -Wno-error=implicit-int   -Wno-error=unknown-warning-option   -Wno-error=vla-cxx-extension   -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ -O1   -fno-omit-frame-pointer   -gline-tables-only   -Wno-error=incompatible-function-pointer-types   -Wno-error=int-conversion   -Wno-error=deprecated-declarations   -Wno-error=implicit-function-declaration   -Wno-error=implicit-int   -Wno-error=unknown-warning-option   -Wno-error=vla-cxx-extension   -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ tests/capi/CMakeFiles/torture_test.dir/torture_test.cc.o -o tests/capi/torture_test  lua-master/source/lib
[54/66] cd /src/testdir/build/luzer/build && cd /src/testdir/build/luzer/build && /usr/bin/ninja && /usr/local/bin/cmake -E touch /src/testdir/build/luzer/stamp/bundled-luzer-build
FAILED: luzer/stamp/bundled-luzer-build luzer/build/luzer/luzer/luzer.so luzer/build/luzer/luzer/libcustom_mutator.so /src/testdir/build/luzer/stamp/bundled-luzer-build /src/testdir/build/luzer/build/luzer/luzer/luzer.so /src/testdir/build/luzer/build/luzer/luzer/libcustom_mutator.so 
cd /src/testdir/build/luzer/build && cd /src/testdir/build/luzer/build && /usr/bin/ninja && /usr/local/bin/cmake -E touch /src/testdir/build/luzer/stamp/bundled-luzer-build
[1/16] Building C object luzer/CMakeFiles/luzer_impl.dir/compat.c.o
[2/16] Building C object luzer/CMakeFiles/luzer_impl.dir/tracer.c.o
[3/16] Building C object luzer/CMakeFiles/luzer_impl.dir/metrics.c.o
[4/16] Building C object luzer/CMakeFiles/luzer_impl.dir/counters.c.o
[5/16] Building C object luzer/CMakeFiles/luzer_impl.dir/config.c.o
[6/16] Copy libFuzzer and sanitizer libraries
[7/16] Strip sanitizer library ubsan
[8/16] Build sanitizer library ubsan
FAILED: luzer/CMakeFiles/build_dso_ubsan luzer/libfuzzer_with_ubsan.so /src/testdir/build/luzer/build/luzer/CMakeFiles/build_dso_ubsan /src/testdir/build/luzer/build/luzer/libfuzzer_with_ubsan.so 
cd /src/testdir/build/luzer/build/luzer && /usr/local/bin/clang -Wl,--whole-archive libclang_rt.fuzzer_no_main.a libclang_rt.ubsan_standalone.a -Wl,--no-whole-archive -lstdc++ -lpthread -ldl -shared -o libfuzzer_with_ubsan.so
clang: error: no such file or directory: 'libclang_rt.fuzzer_no_main.a'
[9/16] Copy libFuzzer and sanitizer libraries
[10/16] Building C object luzer/CMakeFiles/custom_mutator.dir/custom_mutator_lib.c.o
[11/16] Building C object luzer/CMakeFiles/luzer_impl.dir/luzer.c.o
[12/16] Building CXX object luzer/CMakeFiles/luzer_impl.dir/fuzzed_data_provider.cc.o
ninja: build stopped: subcommand failed.
```

Version: aae4bbd
Job: https://github.com/google/oss-fuzz/actions/runs/25180793834/job/73825186615?pr=15442

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -57,3 +57,4 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - A memory leak in a Lua-based implementation of `TestOneInput()`.
 - An initial buffer size in FuzzedDataProvider.
 - Search of the archiver tool (`ar`) in the OSS Fuzz environment.
+- An error when Clang couldn't find the library.

--- a/cmake/BuildLibSanitizers.cmake
+++ b/cmake/BuildLibSanitizers.cmake
@@ -40,8 +40,8 @@ macro(GEN_BUILD_TARGET name libsanitizer_path libfuzzer_path
   set(LINK_COMMAND
     ${CMAKE_C_COMPILER}
     -Wl,--whole-archive
-    ${libfuzzer_name}
-    ${libsanitizer_name}
+    ${CMAKE_CURRENT_BINARY_DIR}/${libfuzzer_name}
+    ${CMAKE_CURRENT_BINARY_DIR}/${libsanitizer_name}
     -Wl,--no-whole-archive
     -lstdc++
     -lpthread
@@ -52,8 +52,8 @@ macro(GEN_BUILD_TARGET name libsanitizer_path libfuzzer_path
   if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     set(LINK_COMMAND
       ${CMAKE_C_COMPILER}
-      ${libfuzzer_name}
-      ${libsanitizer_name}
+      ${CMAKE_CURRENT_BINARY_DIR}/${libfuzzer_name}
+      ${CMAKE_CURRENT_BINARY_DIR}/${libsanitizer_name}
       -lstdc++
       -lpthread
       -dynamiclib

